File: scripts/prepare-public.py
#!/usr/bin/env python3
"""Stage only explicitly approved public files for the production build."""
import shutil
from assets import ROOT, read_json, verify_installed

lock = read_json(ROOT / "assets.lock.json")
verify_installed(lock, ROOT / "public/factorio")
output = ROOT / ".local/build-public"
if output.exists():
    shutil.rmtree(output)
for name in lock["files"]:
    destination = output / "factorio" / name
    destination.parent.mkdir(parents=True, exist_ok=True)
    shutil.copyfile(ROOT / "public/factorio" / name, destination)
branding = output / "branding"
branding.mkdir(parents=True)
shutil.copyfile(ROOT / "public/branding/josh-cazalas.png", branding / "josh-cazalas.png")
print(f"Staged {len(lock['files'])} runtime assets and the nameplate")

File: scripts/prepare-public.ts
import { copyFileSync, mkdirSync, rmSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { ROOT, readJson, writeJson } from './lib/artifacts.ts';
import { buildIdentity } from './lib/build-identity.ts';
import { verifyInstalled } from './assets.ts';
import type { AssetLock } from './assets.ts';

const lock = readJson<AssetLock>(join(ROOT, 'assets.lock.json'));
verifyInstalled(lock, join(ROOT, 'public/factorio'));
const output = join(ROOT, '.local/build-public');
rmSync(output, { recursive: true, force: true });
for (const name of Object.keys(lock.files)) {
  const destination = join(output, 'factorio', name);
  mkdirSync(dirname(destination), { recursive: true });
  copyFileSync(join(ROOT, 'public/factorio', name), destination);
}
mkdirSync(join(output, 'branding'), { recursive: true });
copyFileSync(join(ROOT, 'public/branding/josh-cazalas.png'), join(output, 'branding/josh-cazalas.png'));
writeJson(join(output, 'release.json'), buildIdentity());
console.log(`Staged ${Object.keys(lock.files).length} runtime assets and the nameplate`);
